src/application/clients/crux-wallet-client.ts

```diff
@@ -0,0 +1,211 @@
+import { ICruxBlockstackInfrastructure } from "../../core/interfaces";
+import { ICruxUserRepository } from "../../core/interfaces/crux-user-repository";
+import { IKeyManager } from "../../core/interfaces/key-manager";
+import { ICruxIDState, setCacheStorage } from "../../index";
+import { BasicKeyManager } from "../../infrastructure/implementations/basic-key-manager";
+import { BlockstackCruxDomainRepository } from "../../infrastructure/implementations/blockstack-crux-domain-repository";
+import { BlockstackCruxUserRepository } from "../../infrastructure/implementations/blockstack-crux-user-repository";
+import { BaseError, CruxClientError, ErrorHelper, PackageErrorCode } from "../../packages/error";
+import { CruxDomainId, CruxId, InputIDComponents } from "../../packages/identity-utils";
+import { InMemStorage } from "../../packages/inmem-storage";
+import { StorageService } from "../../packages/storage";
+import { cloneValue } from "../../packages/utils";
+
+export const throwCruxClientError = (target: any, prop: any, descriptor?: { value?: any; }): any => {
+    let fn: any;
+    let patchedFn: any;
+    if (descriptor) {
+        fn = descriptor.value;
+    }
+    return {
+        configurable: true,
+        enumerable: true,
+        get() {
+            if (!patchedFn) {
+                patchedFn = async (...params: any[]) => {
+                    try {
+                        return await fn.call(this, ...params);
+                    } catch (error) {
+                        throw CruxClientError.fromError(error);
+                    }
+                };
+            }
+            return patchedFn;
+        },
+        set(newFn: any) {
+            patchedFn = undefined;
+            fn = newFn;
+        },
+    };
+};
+export interface ICruxWalletClientOptions {
+    privateKey?: string;
+    blockstackInfrastructure?: ICruxBlockstackInfrastructure;
+    cacheStorage?: StorageService;
+    walletClientName: string;
+}
+
+export class CruxWalletClient {
+    public walletClientName: string;
+    private cruxBlockstackInfrastructure: ICruxBlockstackInfrastructure;
+    private _initPromise: Promise<void>;
+    private _cruxUser?: CruxUser;
+    private cruxDomain?: CruxDomain;
+    private _cruxUserRepository: ICruxUserRepository;
+    private _cruxAssetTranslator?: CruxAssetTranslator;
+    private _keyManager?: IKeyManager;
+    private resolvedClientAssetMapping?: IResolvedClientAssetMap;
+    private cacheStorage?: StorageService;
+
+    constructor(options: ICruxWalletClientOptions) {
+        setCacheStorage(options.cacheStorage || new InMemStorage());
+        this.cruxBlockstackInfrastructure = options.blockstackInfrastructure || CruxSpec.blockstack.infrastructure;
+        this.walletClientName = options.walletClientName;
+        this._cruxUserRepository = new BlockstackCruxUserRepository({cacheStorage: this.cacheStorage, blockstackInfrastructure: this.cruxBlockstackInfrastructure});
```

Comment:
there could be some possible overrides from CruxDomain. Ideally the construction of the CruxUserRepository should be inside init method after the construction of CruxDomain. Also the overrides (bnsOverrides?) are not being passed here. Please validate.

Reply:
Moved the construction of the CruxUserRepository to _init after the construction of CruxDomain and passing bnsNodes from cruxDomain config to CruxUserRepository as bnsOverrides